docs/proposals/portable-service-assets.md

```diff
@@ -0,0 +1,84 @@
+  documentationUrl: "{{ assets }}/docs.html"
+```
+
+## Downloading Assets
+After every bootstrap of apb specs, ASB will begin downloading assets in an asynchronous work job.  For each spec, ASB will check to see if there is an associated assets image ID.  If there is none, no assets are downloaded.  If there is an assets image ID and the assets are missing or the ID has changed from the last download, it will download them from the built image assets (e.g. `hello-world-apb:_assets_latest`) using commands mimicking the following:
+```bash
+mkdir -p <www-docroot>/assets/apbs/dh-ansibleplaybookbundle-hello-world-apb-latest/
+docker create docker.io/ansibleplaybookbundle/hello-world-apb:_assets_latest
+docker cp $DOWNLOADED_IMAGE_ID:/assets/. <www-docroot>/assets/apbs/dh-ansibleplaybookbundle-hello-world-apb-latest
+```
+To avoid using the docker daemon, will attempt to use [buildah](https://github.com/projectatomic/buildah) or [skopeo](https://github.com/projectatomic/skopeo) to perform those commands. 
+
+## Serving Assets
+ASB will serve files from a document root.  This folder would be specified by the broker config and defaulted to `/var/www`. This folder should be available without authentication.
+
+The catalog request will be modified to supply the path to assets. The `{{ assets }}` string in the spec can be replaced with the URL for the assets for that specific APB as part of the catalog request, `<asb-route>/public/assets/apbs/<spec-fq-name>`.  To do this we would need to look up the the route to ASB and add the information specific to each spec.  We can look for a route passed in using the broker config.  If missing, we can try to query for it on some best guesses.
+```bash
+# route name provided:
+oc get routes <brokerConfig.RouteName> -n <brokerConfig.Namespace> -o custom-columns=host:.spec.host --no-headers
+# or if no route name, assume service is named asb:
+oc get routes -n <brokerConfig.Namespace> -l service=asb -o custom-columns=host:.spec.host --no-headers
+```
+
+The resulting request would look like:
+```json
+[
+     {
+          "name": "dh-ansibleplaybookbundle-hello-world-apb-latest",
+          "metadata": {
+             "imageUrl": "https://asb-1338-ansible-service-broker.172.17.0.1.nip.io/public/assets/apbs/dh-ansibleplaybookbundle-hello-world-apb-latest/my-pic.svg",
+             "documentationUrl": "https://asb-1338-ansible-service-broker.172.17.0.1.nip.io/public/assets/apbs/dh-ansibleplaybookbundle-hello-world-apb-latest/docs.html"
+         }
+     }
+ ]
+```
+This will then be used by the service catalog as any normal URI and the asset will be served from ASB.
+
+## Work Items
+* APB tooling to build assets
+* Serving files
+* Downloading assets
+* Modify APB processing for requests
+* Update documentation for building APBs
+
+## Future Considerations
+For running multiple instances of the broker, we'll have to eventually use a persistent volume and consider multiple pods downloading assets.
```

Comment:
It would be interesting if creating a pod with a httpd server on it dedicated to serve content would be useful or not.